sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -310,21 +314,39 @@ private Optional<SclReportItem> updateSourceDataSetsAndControlBlocks(TExtRef ext
 
         try {
             sourceDas.forEach(sourceDa -> {
-                String dataSetName = generateDataSetName(extRef, sourceDa, isBayInternal);
-                ServiceSettingsType serviceSettingsType = ServiceSettingsType.fromTServiceType(extRef.getServiceType());
-                DataSetAdapter dataSet = sourceLDevice.getLN0Adapter().createDataSetIfNotExists(dataSetName, serviceSettingsType);
-                createFCDAInDataSet(extRef, sourceDa, dataSet);
+                String datasetSuffix = generateDataSetSuffix(extRef, sourceDa, isBayInternal);
+                createDataSetWithFCDA(extRef, sourceLDevice, sourceDa, datasetSuffix);
+                createControlBlockWithTarget(extRef, sourceLDevice, sourceDa, datasetSuffix);
             });
         } catch (ScdException e) {
-            // ScdException can be thrown if AccessPoint does not have DataSet creation capability
+            // ScdException can be thrown if AccessPoint does not have DataSet/ControlBlock creation capability
             log.error(e.getMessage(), e);
             return fatalReportItem(extRef, "Could not create DataSet for this ExtRef : " + e.getMessage());
         }
         return Optional.empty();
     }
 
-    private static String generateDataSetName(TExtRef extRef, ResumedDataTemplate sourceDa, boolean isBayInternal) {
-        return "DS_" + extRef.getLdInst() + "_"
+    private void createDataSetWithFCDA(TExtRef extRef, LDeviceAdapter sourceLDevice, ResumedDataTemplate sourceDa, String datasetSuffix) {
+        DataSetAdapter dataSetAdapter = sourceLDevice.getLN0Adapter().createDataSetIfNotExists("DS_" + datasetSuffix, ControlBlockEnum.from(extRef.getServiceType()));
+        dataSetAdapter.createFCDAIfNotExists(extRef.getLdInst(), extRef.getPrefix(), extRef.getLnClass().stream().findFirst().orElse(null), extRef.getLnInst(),
+            sourceDa.getDoRef(),
+            (extRef.getServiceType() == TServiceType.REPORT ? null : sourceDa.getDaRef()),
+            sourceDa.getFc());
+    }
+
+    private void createControlBlockWithTarget(TExtRef extRef, LDeviceAdapter sourceLDevice, ResumedDataTemplate sourceDa, String datasetSuffix) {
+        String cbName = "CB_" + datasetSuffix;
+        String cbId = generateControlBlockId(cbName, sourceLDevice.getLdName(), getParentAdapter());
+        ControlBlockAdapter controlBlockAdapter = sourceLDevice.getLN0Adapter().createControlBlockIfNotExists(cbName, cbId, "DS_" + datasetSuffix, ControlBlockEnum.from(extRef.getServiceType()));
+        if (sourceDa.getFc() != TFCEnum.ST && controlBlockAdapter.getCurrentElem() instanceof TReportControl tReportControl){
+            tReportControl.getTrgOps().setDchg(false);
+            tReportControl.getTrgOps().setQchg(false);
```

Comment:
Not needed!
The default value for both `dchg` and `qchg` is false. 
[dchg : SCL2007B4/SCL_IED.xsd#L416](https://github.com/com-pas/compas-scl-xsd/blob/0.0.4/src/main/resources/xsd/SCL2007B4/SCL_IED.xsd#L416)
[qchg:  SCL2007B4/SCL_IED.xsd#L417](https://github.com/com-pas/compas-scl-xsd/blob/0.0.4/src/main/resources/xsd/SCL2007B4/SCL_IED.xsd#L417)

Reply:
This is true for the XSD, but in our case `createControlBlockIfNotExists` takes the default value from class `ReportControlBlock` which set them to true (because it is the default for TFCEnum.ST).
I set all the defaults values in the constructor of the DTO class `ReportControlBlock`, `GooseControlBlock`, `SMVControlBlock`.
And then I use the mapping method `toControl()` to map them to JAXB classes (`TReportControl`, `TGSEControl`, `TSampledValueControl`).

```java
public class ReportControlBlock extends ControlBlock {
...
    protected TTrgOps trgOps = newDefaultTrgOps();
...  
    private static TTrgOps newDefaultTrgOps() {
        TTrgOps newDefaultTrgOps = new TTrgOps();
        newDefaultTrgOps.setDchg(true);
        newDefaultTrgOps.setQchg(true);
        newDefaultTrgOps.setDupd(false);
        newDefaultTrgOps.setPeriod(true);
        newDefaultTrgOps.setGi(true);
        return newDefaultTrgOps;
    }
```

Reply:
For me, it makes more sense, however in https://github.com/com-pas/compas-sct/issues/85  there is no mention that the default value for `TFCEnum.ST` is true, all I can make out is that it is true.

Reply:
OK. I asked Damien to update the issue.